reportportal_client/core/test_manager.py

```diff
@@ -0,0 +1,221 @@
+                                        name,
+                                        item_type,
+                                        self.launch_id,
+                                        uuid,
+                                        **item_data)
+        test_item.start(start_time)
+        return uuid
+
+    def update_test_item(self, item_uuid, attributes=None, description=None,
+                         **kwargs):
+        """Update existing test item at the Report Portal.
+
+        :param str item_uuid:   test item UUID returned on the item start
+        :param str description: test item description
+        :param dict attributes: test item attributes(tags)
+                                Pairs of key and value (see class doc string)
+        """
+        self.get_test_item(item_uuid)
+        raise NotImplementedError()
+
+    def finish_test_item(self,
+                         item_uuid,
+                         end_time,
+                         status,
+                         issue=None,
+                         attributes=None,
+                         **kwargs):
+        """Finish test item.
+
+        :param item_uuid:  id of the test item
+        :param end_time:   time in UTC format
+        :param status:     status of the test
+        :param issue:      description of an issue
+        :param attributes: dict with attributes
+        :param kwargs:     other parameters
+        """
+        # check if the test is skipped, if not - do not mark as TO INVESTIGATE
+        if issue is None and status == "SKIPPED":
+            issue = {"issue_type": "NOT_ISSUE"}
+        if attributes and isinstance(attributes, dict):
+            attributes = dict_to_payload(attributes)
+        self.get_test_item(item_uuid).finish(end_time, status, issue=issue,
+                                             attributes=attributes, **kwargs)
+
+    def remove_test_item(self, item_uuid):
+        """Remove test item by uuid.
+
+        :param item_uuid: test item uuid
+        """
+        self.get_test_item(item_uuid)
+        raise NotImplementedError()
+
+    def log(self, time, message=None, level=None, attachment=None,
+            item_id=None):
+        """Log message. Can be added to test item in any state.
+
+        :param time:        log time
+        :param message:     log message
+        :param level:       log level
+        :param attachment:  attachments to log (images,files,etc.)
+        :param item_id:     parent item UUID
+        :return:            log item UUID
+        """
+        uuid = generate_uuid()
+        # Todo: Do we store log items?
+        log_item = RPLogItem(self.rp_url,
+                             self.session,
+                             self.api_version,
+                             self.project_name,
+                             self.launch_id,
+                             uuid)
+        log_item.create(time, attachment, item_id, level, message)
```

Comment:
`attachment` is not supported by the RPLogItem.create method.

Reply:
If I understand correct , Serghei implemented attachments as an file_obj. Does it work or not I do not know since those code was not implemented by me.